### PR TITLE
Added heartbeat job to monitor the health of sidekiq/cron

### PR DESCRIPTION
**Pull request by askldjd**
This PR is an attempt to get a better understand on issue https://github.com/department-of-veterans-affairs/caseflow/issues/1814.

It adds a heartbeat job that forces Sidekiq Cron to schedule a job every 5 min. The hope is to determine at exactly when, and how the jobs stops running.

This PR does not resolve the issue by itself. The change set will be removed once we figure out the bug.

- Adds Sidekiq UI for useful debugging info

### CODE REVIEW

File: config/sidekiq_cron.yml [hunk 1]
```diff
@@ -17,3 +18,12 @@ prepare_establish_claim_tasks_job:
   class: "PrepareEstablishClaimTasksJob"
   queue: default
   active_job: true
+
+# This job is used to debug https://github.com/department-of-veterans-affairs/caseflow/issues/1814
+# It is scheduled to run every 5 minute to make sure Sidekiq and Sidekiq Cron
+# are both operational. It will be removed once the issue is fixed.
+heartbeat_tasks_job:
+  cron: "*/5 * * * * America/New_York"
```
Comment (joofsh) [inline config/sidekiq_cron.yml]: the comment says every 5 min (which makes sense to me). Adjusted the cron accordingly. It was previously every minute

File: config/sidekiq_cron.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ create_establish_claim_tasks_job:
   cron: "1 0 * * * America/New_York"
   class: "CreateEstablishClaimTasksJob"
   queue: default
+  active_job: true
```
Comment (joofsh) [inline config/sidekiq_cron.yml]: this was the only inconsistency in the setups that I could find. So `CreateEstablishClaimTasksJob` was going through a very different code path than the other jobs. This _might_ be responsible for the issues we've seen (though I can't understand why it'd succeed sometimes)

**Comment by joofsh** [comment]
**Sidekiq UI**
screen shot 2017-05-02 at 3 37 02 pm

**Sidekiq Cron jobs**

- A way for the support team to trigger the jobs if they get a support ticket saying there are no tasks. This helps avoid bugging a developer urgently
- Avoids  going into the production rails console (better security)
- Shows which cron jobs are active (if we see one disappear from the UI or become disabled, that's bad)
- Shows when the cron job was last run (we would expect it to run every night)
screen shot 2017-05-02 at 3 37 07 pm





See related efolder PR: https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/476

## Next Steps for debugging:
- Sometime should take ownership of manually checking the jobs each morning. Maybe @shanear since you've been doing this already
- If the jobs did not run overnight. **BEFORE FIXING**, they should check the sidekiq cron UI and see if:
  - Cron job exists and is enabled
  - When it was last run


CC @shanear @amprokop for awareness

File: config/routes.rb [hunk 3]
```diff
@@ -113,5 +113,19 @@
       post "/set-end-products", to: "users#set_end_products", as: 'set_end_products'
     end
   end
+
+  require "sidekiq/web"
+  require "sidekiq/cron/web"
+  Sidekiq::Web.use Rack::Auth::Basic do |username, password|
```
Comment (askldjd) [inline config/routes.rb]: Awesome. I saw this, but wasn't sure how useful it is to debug sidekiq issues.

Love to see this in action.

File: config/sidekiq_cron.yml (hunk 1 above)
Comment (askldjd) [inline config/sidekiq_cron.yml]: :+1:

File: config/sidekiq_cron.yml (hunk 2 above)
Comment (askldjd) [inline config/sidekiq_cron.yml]: I saw that too. Since it had worked before, I sort of ignored it.

**Comment by askldjd** [comment]
So... I can't approve my own PR. Looks like you gotta approve it. :expressionless:

**Comment by askldjd** [comment]
:+1: for the "Next Steps for Debugging".

I would also recommend to look over the Sidekiq Cron and Sidekiq Retry/Dead Queue in the rails console.

https://github.com/ondrejbartas/sidekiq-cron
https://github.com/mperham/sidekiq/wiki/API

**Pull request merged**